src/Shape.cpp

```diff
@@ -21,12 +21,12 @@ Shape* Shape::clone() const {
     return new Shape(*this);
 }
 
-bool Shape::intersect(const Ray &, float t0, float t1, double& distance) const {
+bool Shape::intersect(const Ray& /*r*/, float /*t0*/, float /*t1*/, double& /*distance*/) const {
     return false;
 }
```

Comment:
This function was implemented pretty strangely before don't you think? There is no implementation to determine the intersect between a regular shape and ray so we just return false. Should we give a not-implemented warning here?

Reply:
Sounds like a plan.